### ISSUE TITLE
Can't drag directories onto the command line

### ISSUE THREAD

**Issue by skllcrn** [zeit/hyper#3582]
- **OS version and name**: macOS 10.14.4
- **Hyper.app version**: Hyper 3.0.0-canary.8 (stable)

## Issue
![Kapture 2019-05-06 at 1 21 43](https://user-images.githubusercontent.com/8440026/57201764-b93f6900-6f9d-11e9-8f3b-307ceb754a07.gif)

**Comment by xavi-dev** [comment]
Same here with satable version: Hyper 3.0.0 (stable)

**Comment by ZaLiTHkA** [comment]
same thing with `3.0.0` stable on Windows 10 x64, doesn't seem to matter what `shell` I use (normally `bash.exe` from my Git program folder). pretty sure this worked in `2.x`...

**Comment by robinscholz** [comment]
same issue

**Comment by anthonylauzon** [comment]
Same issue

**Comment by jmmaynard** [comment]
Same issue here.

**Comment by bgold0** [comment]
This doesnt work in 3.0.0 or 3.0.2

**Comment by KaKaVip** [comment]
Same issue

**Comment by tam-lam** [comment]
Hyper 3.0.2 (stable) 
Same issue, very inconvenient ,properly stop using this version until the issue is fixed

**Comment by cansik** [comment]
Really sad that the support for this feature is not back already. This definitely makes hyper useless for me; usually the "copy&paste" work around is not possible, because something else is already in the clipboard.

**Comment by Koenkk** [comment]
Same here on 3.0.2 (Windows)

**Comment by nhterry** [comment]
Ditto. 

3.0.2 (Stable) Windows 10

**Comment by keptup** [comment]
Copy-pasting a folder to terminal using keyboard still works. A temporary workaround I guess.

**Comment by skibitsky** [comment]
Same on 3.0.2 on macOS

**Comment by luxcium** [comment]
ha ok I am not the only one I hope it will be back soon (either I am too lazy to write the full path or it's just about being more productive I don't know)

on 3.0.2 on macOS 10.14.4
Renderers: Canvas
Plugins: 
  hyper-opacity (1.0.0),
  hyperlayout (0.2.1)

**Comment by xavi-dev** [comment]
Smalll workarround on macOS: **CMD+OPTION+C** This will copy the path of the file.

If you are better with mouse, then right click on the file and press option to see how the "copy file" changes to "copy file as path"

**Comment by rodolfoviolac** [comment]
+1!
Same problem on mac 10.14.4 with hyper 3.0.2

**Comment by kubatruhlar** [comment]
What a shame the feature was removed.

**Comment by architchandra** [comment]
I'm facing the same problem. Please fix this.

Mac OS: 10.14.5
Hyper: 3.0.2 (stable)

**Comment by igoticecream** [comment]
Same issue

**Comment by skibitsky** [comment]
Switching to iTerm2 fixed for me

With best regards,
Gleb Skibitsky

**Comment by skllcrn** [comment]
Please use 👍 to add your voice to the issue, unless you have additional information to add. Thank you!

**Comment by echobb8** [comment]
If there is a space in the name of a file/folder, right-clicking and choosing the option to "copy file/folder as path" is not working, because it's not adding a backslash after the words with spaces that come after them. 

Example: 

$ cd /Users/user/Documents/Test Folder/subfolder
bash: cd: /Users/user/Documents/Test: No such file or directory
$ cd /Users/user/Documents/Test\ Folder/subfolder (Manually added the backslash after "Test")
hostname:subfolder user$

TL;DR:
Fix this bug.

**Comment by adeonir** [comment]
Smalll workarround on macOS: CMD+C on Finder and CMD+V on Hyper.
This will paste the path with single quotes.

**Comment by slang25** [comment]
Thanks for addressing this @chabou 🎉

**Comment by architchandra** [comment]
It's mentioned here that the issue is closed but I'm on the latest available version and this still does not work.

MacOS version: Mojave v10.14.6 (18G87)
Hyper version: 3.0.2 (stable)
Plugins used: hyper-snazzy (1.3.0)

**Comment by bet4it** [comment]
@architchandra The solution hasn't been included in any public releases yet. If you want to try it, you need to build `hyper` from source.

**Comment by architchandra** [comment]
Ah. Any expected timelines?

**Comment by eccoscreen** [comment]
Is this going to be fixed? This feature is pretty much a deal breaker, I've since moved to using the regular terminal app on macOS.

**Comment by mrpbennett** [comment]
> Is this going to be fixed? This feature is pretty much a deal breaker, I've since moved to using the regular terminal app on macOS.

I agree, I have switched back to what I was previously using. I love hyper for the easy of use but this sucks.

**Comment by jamesaspence** [comment]
Does anybody know when this fix will be released? I found the commit from late July that appears to resolve it, but no release since May.

**Comment by ahorvat** [comment]
Very confused why the issue is closed when the fix is not released

**Comment by mrpbennett** [comment]
I have switched back to iterm2 because of this.

**Comment by architchandra** [comment]
If I build from source, would Hyper be able to detect when a public release is out and prompt me to update/update automatically? Or will I have to keep an eye out for when the public release happens?

**Comment by warpling** [comment]
@mrpbennett @mrpbennett @eccoscreen and others in this thread who are whining: Yes it's an annoying bug but take the attitude elsewhere. Hyper is a free project maintained by volunteers.  What a thankless job.

**Comment by qweasd1** [comment]
guys, I create a plugin for this: https://github.com/qweasd1/hyper-drop-file, just  ```hyper install hyper-drop-file```

**Comment by ruucm** [comment]
@qweasd1  cool! though, it doesn't work with zsh?

**Comment by cansik** [comment]
As others already mentioned, it is quite easy to build hyper on your own and get a fix for this. The main problem I have now is, that I already adapted my workflow to always copy the file/directory and past it into hyper...and this is not supported anymore in the newest build.

So this "bug" forced me to learn something new and now I have to readapt because it is fixed. This leads to a not so great user experience, it would have been better to keep both ways on how to drag / copy a path into the command line (for those who adapted, and for the others).

**Comment by qweasd1** [comment]
> @qweasd1 cool! though, it doesn't work with zsh?

Hi ruucm, I test my plugin with zsh, it works fine.
I follows the step here to use zsh in hyper: https://tjay.dev/howto-my-terminal-shell-setup-hyper-js-zsh-starship/ and I test it with my plugin, it works fine on my mac. 

So if it doesn't work for you, feel free to let me know the details, I will see if I could help!

**Comment by ruucm** [comment]
@qweasd1 

Thanks for the reply. when I reinstall it. it works well :) 

Thanks, man!

**Comment by qweasd1** [comment]
sounds great!

**Comment by machieajones** [comment]
> guys, I create a plugin for this: https://github.com/qweasd1/hyper-drop-file, just `hyper install hyper-drop-file` and _DO RESTART YOUR TERMINAL_ for the first time use. If you find it's useful, please star this plugin, thanks!

Thank you for this!

**Comment by LabhanshAgrawal** [comment]
Please test the build from #5781 and confirm if it's fixed.
https://github.com/vercel/hyper/actions/runs/1051719012#artifacts
Can be included in next patch release, if working.
